### ISSUE TITLE
Reducing an image, but increasing the size of the file

### ISSUE THREAD

**Issue by pattousai** [jcupitt/libvips#704]
I'm running some tests in the PaSERC benchmark and i need to reduce the image that have a 18000x18000 size. I use the resize function (vips resize input.v output.v 0,9 or using the resize in python or using the resize in C++) with a 0,9 scale but the output file size increase a lot (original: 1,2 GB, output with a 0,9 scale: 2,9 GB).

Why this is happening? How can i reduce the image size, reducing the file size by the same proportion?

Thanks.

**Comment by jcupitt** [comment]
Hello, maybe try "0.9"? I see:

```
john@kiwi:~/pics$ ls -l wtc.v
-rw-r--r-- 1 john john 263509456 Aug  3 14:19 wtc.v
john@kiwi:~/pics$ vips resize wtc.v wtc2.v 0.9
john@kiwi:~/pics$ ls -l wtc2.v
-rw-r--r-- 1 john john 213453979 Aug  3 14:26 wtc2.v
```

What type of input file are you using? I see:

```
john@kiwi:~/pics$ vipsheader wtc.v
wtc.v: 9372x9372 uchar, 3 bands, srgb, jpegload
```

(the jpegload just means this image was originally from a jpeg source, the file itself is obviously in vips format)

**Comment by pattousai** [comment]
vipsheader give me the following:

`orion_18000x18000.v: 18000x18000, labq`

actually the syntax in my computer is "0,9", when i use "0.9" give me the error:

```
(vips:16965): GLib-GObject-WARNING **: value "inf" of type 'gdouble' is invalid or out of range for property 'vshrink' of type 'gdouble'
reducev: parameter vshrink not set
```

But i have the same result if a use some python code, just as this:

```python
image = Vips.Image.new_from_file("orion_18000x18000.v")
new_image = image.resize(0.9)
new_image.write_to_file("output.v")
```

Or the C code:

```
#include <stdio.h>

#include <vips/vips.h>
 
int
main (int argc, char **argv)
{
  VipsImage *im;
  VipsImage *out;
  double scale;
 
  if (VIPS_INIT (argv[0]))
    vips_error_exit ("unable to start VIPS");

  if (argc != 2)
    vips_error_exit ("usage: %s <filename>", g_get_prgname ());

  if (!(im = vips_image_new_from_file (argv[1], NULL)))
    vips_error_exit ("unable to open");
  
  scale = 0.9;
  
  vips_resize (im, &out, scale, NULL);
  
  vips_image_write_to_file( out, "out.v", NULL);
  
  g_object_unref (im);
  g_object_unref (out);
 
 
  return (0);
}
```

**Comment by jcupitt** [comment]
Ah OK, labq images will be expanded to float during shrink, you need to pack back again for write (if you want to keep labq). In Python, add:

```python
image = image.colourspace("labq")
```

Just before the write.

labq images can't be computed with directly, so all operations (except things like crop, which can work on labq) will automatically unpack them to lab (3 x float). They don't automatically repack to labq afterwards, because in a chain of operations you could have an image being repeatedly packaged and unpacked, which would be horrible for performance. 

In short, labq is special and you'll need to have an extra thing on save if you want to use it.

**Comment by pattousai** [comment]
Man, thanks very much, that's was exactly the "problem".

Surprised with the speed of your answer. Congrats

**Comment by jcupitt** [comment]
I'll close, please reopen if you still have problems.
